google-ads-examples/src/main/java/com/google/ads/googleads/examples/recommendations/DetectAndApplyRecommendations.java

```diff
@@ -116,56 +115,76 @@ private void runExample(GoogleAdsClient googleAdsClient, long customerId)
             googleAdsClient.getLatestVersion().createRecommendationServiceClient()) {
       // Creates a query that retrieves keyword recommendations.
       String query =
-          "SELECT recommendation.resource_name "
+          "SELECT recommendation.resource_name, "
+              + "  recommendation.campaign, "
+              + "  recommendation.keyword_recommendation "
               + "FROM recommendation "
-              + "WHERE recommendation.type = KEYWORD "
-              + "LIMIT "
-              + MAX_RESULT_SIZE;
-      for (int i = 0; i < NUMBER_OF_RUNS; i++) {
-        // Constructs the SearchGoogleAdsStreamRequest.
-        SearchGoogleAdsStreamRequest request =
-            SearchGoogleAdsStreamRequest.newBuilder()
-                .setCustomerId(Long.toString(customerId))
-                .setQuery(query)
-                .build();
-
-        // Issues the search stream request.
-        ServerStream<SearchGoogleAdsStreamResponse> stream =
-            googleAdsServiceClient.searchStreamCallable().call(request);
-
-        // Creates apply operations for all the recommendations found.
-        List<ApplyRecommendationOperation> applyRecommendationOperations = new ArrayList<>();
-        for (SearchGoogleAdsStreamResponse response : stream) {
-          for (GoogleAdsRow googleAdsRow : response.getResultsList()) {
-            applyRecommendationOperations.add(
-                ApplyRecommendationOperation.newBuilder()
-                    .setResourceName(googleAdsRow.getRecommendation().getResourceName())
-                    .build());
-          }
-        }
+              + "WHERE recommendation.type = KEYWORD";
+      // Constructs the SearchGoogleAdsStreamRequest.
+      SearchGoogleAdsStreamRequest request =
+          SearchGoogleAdsStreamRequest.newBuilder()
+              .setCustomerId(Long.toString(customerId))
+              .setQuery(query)
+              .build();
+
+      // Issues the search stream request to detect keyword recommendations that exist for the
+      // customer account.
+      ServerStream<SearchGoogleAdsStreamResponse> stream =
+          googleAdsServiceClient.searchStreamCallable().call(request);
+
+      // Creates apply operations for all the recommendations found.
+      List<ApplyRecommendationOperation> applyRecommendationOperations = new ArrayList<>();
+      for (SearchGoogleAdsStreamResponse response : stream) {
+        for (GoogleAdsRow googleAdsRow : response.getResultsList()) {
+          Recommendation recommendation = googleAdsRow.getRecommendation();
+          System.out.printf(
+              "Keyword recommendation '%s' was found for campaign '%s'%n",
```

Comment:
It looks like you're only replacing one of the substitution variables (for campaign). Do you need to change this for the recommendation resource name as well?